models/confirmation.go

```diff
@@ -88,11 +89,17 @@ var (
 func NewConfirmation(theType Type, templateName TemplateName, creatorId string) (*Confirmation, error) {
 
 	shortKey := ""
+	status := StatusPending
 	var err error = nil
-	if theType == TypePatientPasswordReset {
+
+	switch theType {
```

Comment:
I personally find it weird to have this logic specific to one template type to be added in the generic constructor. I would rather have pushed that in the "forgot.go"